src/algebra/order/monoid.lean

```diff
@@ -114,9 +114,13 @@ class linear_ordered_comm_monoid (α : Type*)
 class zero_le_one_class (α : Type*) [has_zero α] [has_one α] [has_le α] :=
 (zero_le_one : (0 : α) ≤ 1)
 
-@[simp] lemma zero_le_one [has_le α] [has_zero α] [has_one α] [zero_le_one_class α] : (0 : α) ≤ 1 :=
+@[simp] lemma zero_le_one [has_zero α] [has_one α] [has_le α] [zero_le_one_class α] : (0 : α) ≤ 1 :=
 zero_le_one_class.zero_le_one
 
+/- `zero_le_one` with an explicit type argument. -/
```

Comment:
I think you should just make the argument explicit on the original lemma

Reply:
There's hundreds of uses of `zero_le_one`, and only a dozen or so that require the explicit type arg. So I disagree, unless there's a strong argument on the other side.

Reply:
It's really important when you make changes like this to link to existing discussion about this kind of detail, rather than creating a PR with a blank description and leaving the reviewer to guess the context.

Only thanks to watching Kevin's discord, I found that @fpvandoorn asked for this PR (as the separate version, what you have now) in https://github.com/leanprover-community/mathlib/pull/14510#issuecomment-1146592609, while @adomani suggested changing the existing one in https://github.com/leanprover-community/mathlib/pull/14510#issuecomment-1146661350.

I don't feel super strongly; my main objection is the use of `'` to indicate argument explicitness, which I don't think we do very often.